### PR TITLE
Extend logstash.node_stats metricset for logstash_stats stack monitoring data

### PR DESCRIPTION
**Pull request by ycombinator**
This PR extends the `node_stats` metricset in the `logstash` metricbeat module. It teaches it to understand the `xpack.enabled` setting in `modules.d/logstash.yml`. If this setting is set, the metricset indexes `logstash_stats` documents in `.monitoring-logstash-*` indices.

Resolves partially: #7035

### Testing this PR
1. Start up a Logstash node running one or more pipelines.
2. Build Metricbeat with this PR:
   ```
   cd metricbeat
   mage build
   ```

3. Enable the logstash Metricbeat module:

   ```
   metricbeat modules enable logstash
   ```

   _Note:_ `metricbeat modules enable logstash-xpack` _will not work as it has not been implemented yet._

4. Configure the logstash Metricbeat module for X-Pack Monitoring. To do this, edit `modules.d/logstash.yml` and add the following line to it:

   ```
   xpack.enabled: true
   ```

5. Start Metricbeat:
   ```
   metricbeat -e
   ```
6. Query the `.monitoring-logstash-7-mb-*` indices to make sure there are documents with `type: logstash_stats`.
   ```
   GET .monitoring-logstash-7-mb-*/_search?q=type:logstash_stats
   ```
7. Repeat the query in 6. above multiple times over the course of a minute or so. Make sure the number of documents grows over time. New documents of `type:logstash_stats` should be created every 10 seconds (corresponding to the default `period` in `modules.d/logstash.yml`).

### CODE REVIEW

**Comment by ycombinator** [comment]
jenkins, test this

**Comment by ycombinator** [comment]
jenkins, test this

File: metricbeat/module/logstash/node_stats/data_xpack.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/helper/elastic"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+type commonStats struct {
+	Events  map[string]interface{} `json:"events"`
+	JVM     map[string]interface{} `json:"jvm"`
```
Comment (ruflin) [inline metricbeat/module/logstash/node_stats/data_xpack.go]: @felixbarny @axw Just found some more jvm metrics. I wonder if long term this also ties into our metrics discussion: https://github.com/elastic/ecs/issues/474

**Comment by ycombinator** [comment]
> Do you plan to also expand the metrics collected by non-xpack?

@ruflin Not right away. I want to deliberately keep this set minimal to start out. It's easier to add more later than take away ones we don't need. So I'd rather add metrics as/when users request them.

**Comment by ycombinator** [comment]
Chatted with @ruflin off-PR and he's given his LGTM. So I'm going ahead with the merge.

**Pull request merged**